src/github.com/getlantern/flashlight/server/server.go

```diff
@@ -28,15 +28,37 @@ const (
 )
 
 var (
-	log = golog.LoggerFor("flashlight.server")
+	log               = golog.LoggerFor("flashlight.server")
+	frontingProviders = map[string]func(*http.Request) bool{
+		// WARNING: If you add a provider here, keep in mind that Go's http
+		// library normalizes all header names so the first letter of every
+		// dash-separated "word" is uppercase while all others are lowercase.
+		// Also, try and check more than one header to lean on the safe side.
+		"cloudflare": func(req *http.Request) bool {
+			return hasHeader(req, "Cf-Connecting-Ip") || hasHeader(req, "Cf-Ipcountry") || hasHeader(req, "Cf-Ray") || hasHeader(req, "Cf-Visitor")
+		},
+		"cloudfront": func(req *http.Request) bool {
+			return hasHeader(req, "X-Amz-Cf-Id") || headerMatches(req, "User-Agent", "Amazon Cloudfront")
+		},
+	}
```

Comment:
I didn't understand that you were suggesting to block our current front.  I should have said "then if your current front was blocked [by censors] that would leave out the vast majority of users".  I think this is a problem and not a feature.  I don't think we'll have any trouble finding incentives for users to upgrade when it makes sense (e.g., when we get auto-updates out).  Limiting fronts doesn't seem like a particularly good one, since it doesn't affect the day to day experience until Lantern stops working altogether.  We might as well tell them that Lantern will stop working in X days.

TBH I think misdetection because these headers change is a theoretical rather than practical concern.  Although these headers may not be meant for this (and in some cases they seem to be meant for this, e.g., "User-Agent: Amazon Cloudfront"), they are meant for _something_ and stuff will break for millions of customers if these headers get changed willy-nilly.  Since we are checking different headers per CDN, we have some redundancy there; some headers could change and we'd still detect the right CDN.

I'll take a look at what would need to change in the client and if it seems like a small, safe change I'll add these headers.  I don't think they merit a significant or risky change at this point (at the very least, until the full-blown Lantern has auto-updates).